Source/BSGConnectivity.m

```diff
@@ -50,21 +50,19 @@ @interface BSGConnectivity ()
 
 @property(nonatomic, assign) SCNetworkReachabilityRef reachabilityRef;
 @property(nonatomic, strong) dispatch_queue_t serialQueue;
-@property(nonatomic) CallbackBlock callbackBlock;
+@property(nonatomic, copy) CallbackBlock callbackBlock;
 
 @end
 
 @implementation BSGConnectivity
 
 - (instancetype)initWithURL:(NSURL *)url
                 changeBlock:(ConnectivityChange)changeBlock {
-    NSString *hostName = [url absoluteString];
-    SCNetworkReachabilityRef ref =
-        SCNetworkReachabilityCreateWithName(NULL, [hostName UTF8String]);
-
     if (self = [super init]) {
-        _connectivityChangeBlock = changeBlock;
-        self.reachabilityRef = ref;
+        NSString *hostName = [url absoluteString];
+        _reachabilityRef = SCNetworkReachabilityCreateWithName(NULL, [hostName UTF8String]);
+
+        _connectivityChangeBlock = [changeBlock copy];
```

Comment:
`connectivityChangeBlock` property is annotated as copy and thus should be a copy.